llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -2703,11 +2707,12 @@ InstructionCost VPReductionRecipe::computeCost(ElementCount VF,
       CondCost = Ctx.TTI.getCmpSelInstrCost(Instruction::Select, VectorTy,
                                             CondTy, Pred, Ctx.CostKind);
     }
+    auto *PhiType = Ctx.Types.inferScalarType(getChainOp());
+    auto *InputType = Ctx.Types.inferScalarType(getVecOp());
```

Comment:
How is this change related to the current patch? Shouldn't all operands have the same types, as we are passing PR_None?